File: habits/src/main/kotlin/com/akuchars/goals/habits/infrastructure/goal/TemplateQueryServiceImpl.java
package com.akuchars.goals.habits.infrastructure.goal;

import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Service;

import com.akuchars.goals.habits.application.goal.query.TemplateQueryService;
import com.akuchars.goals.habits.domain.goal.model.QTemplate;
import com.akuchars.goals.habits.domain.goal.model.Template;
import com.akuchars.goals.habits.domain.goal.repository.TemplateRepository;

@Service
public class TemplateQueryServiceImpl implements TemplateQueryService {

	@NotNull private final TemplateRepository templateRepository;

	@NotNull private static final QTemplate template = QTemplate.template;


	public TemplateQueryServiceImpl(@NotNull TemplateRepository templateRepository) {
		this.templateRepository = templateRepository;
	}

	@NotNull
	@Override
	public Iterable<Template> findAllByGroupId(long groupId) {
		return templateRepository.findAll(template.group.id.eq(groupId));
	}
}

File: habits/src/main/kotlin/com/akuchars/goals/habits/infrastructure/goal/TemplateQueryServiceImpl.kt
package com.akuchars.goals.habits.infrastructure.goal

import com.akuchars.goals.habits.application.goal.query.TemplateQueryService
import com.akuchars.goals.habits.domain.goal.model.QTemplate
import com.akuchars.goals.habits.domain.goal.model.Template
import com.akuchars.goals.habits.domain.goal.repository.TemplateRepository
import org.springframework.stereotype.Service

@Service
class TemplateQueryServiceImpl (val repository: TemplateRepository): TemplateQueryService {

    private var template: QTemplate = QTemplate.template

    override fun findAllByGroupId(groupId: Long): Iterable<Template> {
        return repository.findAll(template.group.id.eq(groupId))
    }
}